### PR TITLE
Fix VZ guest exec buffering

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary
- Fixes VZ Linux guest exec delivery by preserving the buffered reader used during VSock handshake/readiness when entering command serving.
- Adds and hardens a regression test for the ready ACK and first exec request arriving in the same host write.
- Records the work in Backlog task `TASK-13231`.

## Verification
Rebased onto `dev` at `40345571a2cfc8b3a8893545836097d27e4ee86c`. The buffering defect is still present on that base.

Run from `tools/tldw-agent` with `GOCACHE=/private/tmp/tldw-go-build-cache`:
- `go test -race -count=1 ./...`
- `go vet ./...`
- `go test ./internal/guest -run '^TestGuestVSockClientRunPreservesExecBufferedAfterReady$' -count=20`
- `CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build ./cmd/tldw-agent-guest` (binary emitted to `/private/tmp`)
- Mutation check: a temporary Go overlay restoring the old reader-discarding handoff makes the regression fail after its five-second deadline; the unchanged PR implementation passes.
- `gofmt` check and `git diff --check` pass.
- Independent code review found no actionable Go issues. Both earlier GitHub review threads are resolved.
- Bandit was run from the project virtual environment on the touched guest directory; it found no Python files, so it provides no Go security coverage.

Historical real Apple VZ smoke evidence from the original fix, not rerun during this rebase:
- Rebuilt Debian arm64 bundle; helper daemon bundle smoke: `2 passed`; real host smoke: `3 passed`.
- Evidence: `/private/tmp/tvz-e2e.4iJ3wt/evidence/host-smoke-evidence.json`; `final_exit_code`: `0`.

## Change Summary
This PR fixes the real VZ Linux exec path by preserving the buffered VSock reader after guest handshake/readiness, preventing the first exec request from being lost when it arrives immediately after the ready response. It adds a regression test for that exact buffering edge case and verifies the fix with the Go agent test suite plus a real local Apple VZ smoke run using a rebuilt Debian arm64 bundle.

### CODE REVIEW

File: tools/tldw-agent/internal/guest/vsock_client_test.go [hunk 1]
```diff
@@ -204,21 +207,98 @@ func TestGuestVSockClientReconnectsWithSameVMIDAndToken(t *testing.T) {
...
+		if err != nil {
+			helperDone <- err
+			return
+		}
+
+		execPayload := []byte(`{"protocol_version":"1","request_id":"req-buffered-exec","type":"exec","argv":["/bin/sh","-c","printf preserved > buffered-exec.txt"],"cwd":"."}`)
+		payload := append(append(readyPayload, '\n'), execPayload...)
+		payload = append(payload, '\n')
+		_, err = helperConn.Write(payload)
+		helperDone <- err
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tools/tldw-agent/internal/guest/vsock_client_test.go]: Addressed in 3a0cf8a1d6: the test now waits for and validates the exec response before closing the helper-side pipe, and cancels the fake client context after receiving the expected response.

File: tools/tldw-agent/internal/guest/vsock_client_test.go [hunk 2]
```diff
@@ -204,21 +206,116 @@ func TestGuestVSockClientReconnectsWithSameVMIDAndToken(t *testing.T) {
...
+
+		execPayload := []byte(`{"protocol_version":"1","request_id":"req-buffered-exec","type":"exec","argv":["/bin/echo","preserved"],"cwd":"."}`)
+		payload := append(append(readyPayload, '\n'), execPayload...)
+		payload = append(payload, '\n')
+		if _, err = helperConn.Write(payload); err != nil {
+			helperDone <- err
+			return
+		}
+
+		if err = helperConn.SetReadDeadline(time.Now().Add(time.Second)); err != nil {
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tools/tldw-agent/internal/guest/vsock_client_test.go]: Addressed in 92798edf85: the regression test now uses a 5s read deadline, which keeps the hang bounded without making the test depend on a one-second CI timing assumption.
